### ISSUE TITLE
Columns under 'Output Document Naming' section in the document generator change after generating a document 

### ISSUE THREAD

**Issue by gkahiu** [gltn/stdm#134]
Columns under 'Output Document Naming' section in the document generator change to the those of the first entity in the tab widget even though the current tab widget does not change. This happens immediately after a document has been generated.

See illustration below.

![document generator reset](https://cloud.githubusercontent.com/assets/3840267/17086275/43f0542e-51f6-11e6-9410-f08e875feedc.png)

**Comment by wondie** [comment]
I think this mush have been caused by the resetting of document generator once a document is successfully generated. I don't know why the resetting is needed. I think when it resets the dialog, it restores the first tabs column. 

Will there be any side effects if I don't disable document generator after a document is generated?

**Comment by wondie** [comment]
This issue is now fixed. I have also made few more changes on the doc generation behavior post Generation as listed below. 

- The output box will not change until the user changes the template by going to another tab. If it is in the same tab that template selection is made, it will retain its selection and order. 
- After generation, if the user didn't change the current tab, the output fields and selection will not change. 
- After generation, if successful, the selected records will be removed but if failed or aborted, the selected records will not be removed.
